src/main/scala/com/datastax/gatling/plugin/model/DseCqlAttributes.scala

```diff
@@ -23,32 +26,38 @@ import io.gatling.core.check.Check
   * @param cl               Consistency Level to be used
   * @param cqlChecks        Data-level checks to be run after response is returned
   * @param genericChecks    Low-level checks to be run after response is returned
-  * @param userOrRole       User or role to be used when proxy auth is enabled
```

Comment:
Out of curiosity, what is the recommended way to set up proxy authentication now?

Reply:
Presumably like the retry policy this has now moved to the config... ?

Reply:
Not exactly, I am afraid.

What you are talking about (the config) is when one wants to send all the queries using a proxi'ed user.  It has indeed moved to configuration, as described in the [Proxy Authentication](https://docs.datastax.com/en/developer/java-driver/4.4/manual/core/authentication/#proxy-authentication) section of the driver docs.

This part here is different.  The goal if to give the user a per-query granularity.  See for instance [this RLAC benchmark](https://github.com/riptano/dse-benchmarks/blob/master/src/test/scala/rlac/Rlac_Read_Workload.scala).  We connect to DSE with a `proxyauth` user.  Then, a single Gatling scenario simulates that this `proxyauth` users send query for tends of different users, given it has been granted the `PROXY.EXECUTE` role beforehand.  The goal of this scenario is to figure out whether proxy execution has an overhead.

Think of this as a multi-tenant application, where you want the data of each tenant to not be accessible to anyone else.  You can have a single driver connection, and yet have strict data visibility rules.

Reply:
Thanks for the clarification @pingtimeout .  In a moment of synchronicity I'm also working on porting dse-benchmarks over to use the new driver (see https://github.com/riptano/dse-benchmarks/pull/230/files) and I just happened to hit the RLAC benchmarks just after reading this.  T'is a curious universe, this one.

So... I think the best (only?) mechanism we have for addressing this is exposing the per-request ExecutionProfile settings and requiring users to define profiles for each user role they'll be requesting.  Does that sound right to you @adutra or is there an easier mechanism I'm not aware of?

Reply:
Hang on, I think I can handle this via the ProxyAuthentication class as described in https://docs.datastax.com/en/developer/java-driver/4.4/manual/core/authentication/#proxy-execution.  Working on that now...

Reply:
Okay, I think that resolved it.  @adutra and/or @olim7t would one/both of you mind taking a look at https://github.com/datastax/gatling-dse-plugin/pull/20/commits/cca2ef9ed1f581f041a9f20694c456baa1265aec and confirming that this usage matches what is expected?

Reply:
Yes, that's the way `ProxyAuthentication` is meant to be used. Glad that you didn't have to expose execution-profiles, at least not for now.